lib/sles4sap/sap_deployment_automation_framework/deployment_connector.pm

```diff
@@ -209,15 +212,36 @@ Deployment ID returned from both jobs: 123456 - because it matches with existing
 sub find_deployment_id {
     my (%args) = @_;
     return get_var('SDAF_DEPLOYMENT_ID') if get_var('SDAF_DEPLOYMENT_ID');
+    return get_var('SDAF_GRANDPARENT_ID') if get_var('SDAF_GRANDPARENT_ID');
+
     $args{deployer_resource_group} //= get_required_var('SDAF_DEPLOYER_RESOURCE_GROUP');
     my @check_list = (get_current_job_id(), @{get_parent_ids()});
 
-    diag("Job IDs found: " . Dumper(@check_list));
+    # For SDAF cleanup job it needs its grandparent's id
+    my @parents = get_parent_ids();
+    my $parent_id = $parents[0][0];
+    if (looks_like_number($parent_id) && @parents == 1) {
+        # Get the SDAF deployment job ID, here it is set via SDAF_GRANDPARENT_ID and saved in vars.json
+        my $vars_json = 'vars.json';
+        my $openqa_url = get_required_var('OPENQA_URL');
+        script_run("wget --no-check-certificate -O /tmp/$vars_json https://$openqa_url/tests/$parent_id/file/$vars_json");
```

Comment:
Do we really need to run this on the SUT? It would make more sense to me to run this on the worker.

You could do something like:
```
use Mojo::URL;
use Mojo::UserAgent;
...
my $url = Mojo::URL->new("https://$openqa_url/tests/$parent_id/file/$vars_json") or die "some error";
my $ua = Mojo::UserAgent->new or die "some other error";
$ua->insecure(1);
my $tx = $ua->get($url);
die("Fetching [$vars_json] failed: " . $tx->result->message) unless eval { $tx->result->is_success };
my $json = $tx->res->json;
my $str = $json->{SDAF_GRANDPARENT_ID}, "\n";
```
This way you avoid:

- Possible errors in the SUT.
- Possible errors in the `grep`
- You get a HASHREF with the JSON already parsed
- Nothing is stored anywhere

Reply:
Agree, it is safe at least.
Applied, the new VR is good: https://openqaworker15.qa.suse.cz/tests/321328#step/connect_to_deployer/35
I deleted `die("Fetching [$vars_json] failed: " . $tx->result->message) unless eval { $tx->result->is_success };` as I don't know how to write unit test for it.

Reply:
I believe you can use `dies_ok` and redfine `$ua->get()` to inject some structure for the UT, but even describing it sounds complicated. We can leave it for later.